### PR TITLE
Feature/sdl 0189 restructuring on reset timeout

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Just for internal review

This PR is **[ready]** for review.

Need to test

### CODE REVIEW

File: app/controller/sdl/RController.js [hunk 1]
```diff
@@ -660,6 +660,19 @@ SDL.RController = SDL.SDLController.extend(
...
+     */
+    isEmptyObject: function(object) {
+      var l = 0;
+      for (var key in object) {
+           if(object.hasOwnProperty(key)) {
+               ++l;
+           }
+      }
+      return l == 0;
+   },
```
Comment (vlrm10) [inline app/controller/sdl/RController.js:676]: Better to use :
`return !!Object.keys(object).length`

File: app/controller/sdl/Abstract/Controller.js [hunk 2]
```diff
@@ -691,7 +686,6 @@ SDL.SDLController = Em.Object.extend(
               SDL.AlertPopUp.deactivate();
             }, SDL.AlertPopUp.timeout
           );
-          this.onResetTimeout(element.appID, 'UI.Alert');
```
Comment (vlrm10) [inline app/controller/sdl/Abstract/Controller.js:694]: `UI.Alert` move to const

Comment (vlrm10) [inline app/controller/sdl/Abstract/Controller.js:694]: `UI.Alert` move to const

File: app/controller/sdl/KeyboardController.js [hunk 3]
```diff
@@ -92,9 +92,9 @@ SDL.KeyboardController = Em.Object.create({
      */
     inputChanges: function(element) {
         if (this.appModel && this.appModel.activeRequests.uiPerformInteraction) {
-          SDL.SDLController.onResetTimeout(
-            this.appModel.appID, 'UI.PerformInteraction'
-          );
+          FFW.BasicCommunication.OnResetTimeout(
+            this.requestID, 'UI.PerformInteraction'
```
Comment (vlrm10) [inline app/controller/sdl/KeyboardController.js:96]: `UI.PerformInteraction` move to const

File: app/controller/sdl/KeyboardController.js [hunk 4]
```diff
@@ -148,9 +148,9 @@ SDL.KeyboardController = Em.Object.create({
         }
 
         if (this.appModel && this.appModel.activeRequests.uiPerformInteraction) {
-          SDL.SDLController.onResetTimeout(
-            this.appModel.appID, 'UI.PerformInteraction'
-          );
+          FFW.BasicCommunication.OnResetTimeout(
+            this.requestID, 'UI.PerformInteraction'
```
Comment (vlrm10) [inline app/controller/sdl/KeyboardController.js:152]: same as above

**Comment by vlrm10** [comment]
There are many hardcoded strings in PR. Should be moved to const

File: app/view/sdl/ResetTimeoutPopUp.js [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    },
+
+    /**
+     * setDefaultTimeout function for set default timeout
+     */
+    setDefaultTimeout: function()
+    {
+        self = this;
+        this.resetTimeoutRPCs.forEach(function (method) {
+            if (self.timeoutSeconds[method] == undefined) {
```
Comment (vlrm10) [inline app/view/sdl/ResetTimeoutPopUp.js:169]: Better to use `in` operator to check that the key is defined. Also check all of them in PR

File: app/view/sdl/ResetTimeoutPopUp.js [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+        this.timer = setInterval(
+            function() {
+                    var message = '';
+                    self.resetTimeoutRPCs.forEach(function (method) {
+                        self.timeoutSeconds[method] = self.timeoutSeconds[method] - 1;
+                        message = message + method + ": " + self.timeoutSeconds[method].toString() + '\n';
+                     });
+                    self.set('timeoutString',message);
+            }, 1000
+        );
```
Comment (vlrm10) [inline app/view/sdl/ResetTimeoutPopUp.js:200]: Better to use arrow function in this case

**Comment by LitvinenkoIra** [comment]
Moved to open

**Pull request closed without merging**